src/model/test/HeatExchangerDesiccantBalancedFlow_GTest.cpp

```diff
@@ -0,0 +1,201 @@
+*
+*  (4) Other than as required in clauses (1) and (2), distributions in any form of modifications or other derivative works
+*  may not use the "OpenStudio" trademark, "OS", "os", or any other confusingly similar designation without specific prior
+*  written permission from Alliance for Sustainable Energy, LLC.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include <gtest/gtest.h>
+#include "ModelFixture.hpp"
+
+#include "../HeatExchangerDesiccantBalancedFlow.hpp"
+#include "../HeatExchangerDesiccantBalancedFlow_Impl.hpp"
+
+#include "../HeatExchangerDesiccantBalancedFlowPerformanceDataType1.hpp"
+#include "../HeatExchangerDesiccantBalancedFlowPerformanceDataType1_Impl.hpp"
+#include "../Schedule.hpp"
+#include "../Schedule_Impl.hpp"
+#include "../ScheduleConstant.hpp"
+#include "../ScheduleConstant_Impl.hpp"
+#include "../Node.hpp"
+#include "../Node_Impl.hpp"
+#include "../AirLoopHVAC.hpp"
+#include "../AirLoopHVAC_Impl.hpp"
+#include "../AirLoopHVACOutdoorAirSystem.hpp"
+#include "../AirLoopHVACOutdoorAirSystem_Impl.hpp"
+#include "../HVACTemplates.hpp"
+
+using namespace openstudio;
+using namespace openstudio::model;
+
+TEST_F(ModelFixture, HeatExchangerDesiccantBalancedFlow_HeatExchangerDesiccantBalancedFlow) {
+  ::testing::FLAGS_gtest_death_test_style = "threadsafe";
+
+  ASSERT_EXIT(
+    {
+      Model m;
+      HeatExchangerDesiccantBalancedFlow hx(m);
+      HeatExchangerDesiccantBalancedFlowPerformanceDataType1 p(m);
+      HeatExchangerDesiccantBalancedFlow hx2(m, p);
+
+      exit(0);
+    },
+    ::testing::ExitedWithCode(0), "");
+
+  Model m;
+  HeatExchangerDesiccantBalancedFlow hx(m);
+  HeatExchangerDesiccantBalancedFlowPerformanceDataType1 p(m);
+  HeatExchangerDesiccantBalancedFlow hx2(m, p);
+
+  auto alwaysOn = m.alwaysOnDiscreteSchedule();
+
+  EXPECT_EQ(alwaysOn, hx.availabilitySchedule());
+  ASSERT_TRUE(hx.heatExchangerPerformance().optionalCast<HeatExchangerDesiccantBalancedFlowPerformanceDataType1>());
```

Comment:
heatExchangerPerformance already returns a `HeatExchangerDesiccantBalancedFlowPerformanceDataType1` so it's pointless. instead do

`EXPECT_EQ(p, hx.heatExchangerPerformance());`